test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/ODataUriParserTests.cs

```diff
@@ -1183,6 +1183,94 @@ public void ParseComposableEscapeFunctionUrlAndPropertyReturnsTheSameODataPath()
             Assert.Equal("Name", proSegment.Property.Name);
         }
 
+        [Theory]
+        [InlineData("/entitySetEscaped/32:/photos%2F2018%2F%2fFebruary:/Name", "/entitySetEscaped/32/NS.ComposableFunction(arg='photos%2F2018%2F%2fFebruary')/Name", true)]
+        [InlineData("/entitySetEscaped/32:/photos%2F2018%2F%2fFebruary", "/entitySetEscaped/32/NS.NormalFunction(path='photos%2F2018%2F%2fFebruary')", false)]
+        public void ParseEscapeFunctionWithColonInKeyValue(string escapeFunctionUri, string functionUri, bool isComposable)
+        {
+            // Arrange
+            IEdmModel model = GetEdmModelWithEscapeFunction(escape: true);
+
+            // Act
+            string fullUriString = ServiceRoot + functionUri;
+            ODataUriParser parser = new ODataUriParser(model, ServiceRoot, new Uri(fullUriString));
+            var functionPath = parser.ParsePath();
+
+            fullUriString = ServiceRoot + escapeFunctionUri;
+            parser = new ODataUriParser(model, ServiceRoot, new Uri(fullUriString));
+            var escapePath = parser.ParsePath();
+
+            // Assert
+            Assert.True(functionPath.Equals(escapePath));
+
+            OperationSegment segment = escapePath.First(c => c is OperationSegment) as OperationSegment;
+            Assert.Equal(isComposable ? "NS.ComposableFunction": "NS.NormalFunction", segment.Operations.First().FullName());
+
+            if (isComposable)
+            {
+                PropertySegment proSegment = Assert.IsType<PropertySegment>(escapePath.Last());
+                Assert.Equal("Name", proSegment.Property.Name);
+            }
+        }
+
+        [Theory]
+        [InlineData("/entitySetEscaped/32:/Name", "/entitySetEscaped/32:/Name")]
+        [InlineData("/entitySetEscaped/:32:/Name", "/entitySetEscaped/:32:/Name")]
+        public void ParseKeyValueWithColonWithoutEscapeFunction(string escapeFunctionUri, string functionUri)
+        {
+            // Arrange
+            IEdmModel model = GetEdmModelWithEscapeFunction(escape: false);
+
+            // Act
+            string fullUriString = ServiceRoot + functionUri;
+            ODataUriParser parser = new ODataUriParser(model, ServiceRoot, new Uri(fullUriString));
+            var functionPath = parser.ParsePath();
+
+            fullUriString = ServiceRoot + escapeFunctionUri;
+            parser = new ODataUriParser(model, ServiceRoot, new Uri(fullUriString));
+            var escapePath = parser.ParsePath();
+
+            // Assert
+            Assert.True(functionPath.Equals(escapePath));
+
+            PropertySegment proSegment = Assert.IsType<PropertySegment>(escapePath.Last());

```

Comment:
> Assert.IsType<PropertySegment>(escapePath.Last()); [](start = 40, length = 51)

add the codes to verify the key segment with correct value. for example: the key value has the ":" in .